ddtrace/tracer/span.go

```diff
@@ -625,6 +626,10 @@ func (s *Span) Finish(opts ...FinishOption) {
 		}
 		if cfg.NoDebugStack {
 			s.Lock()
+			if s.finished {
+				s.Unlock()
```

Comment:
pedantic nitpick: We could use s.RLock/s.RUnlock because we only read the value s.finish here, and acquire write lock only we want to `delete`.

Reply:
Done in 1a3f1e77f13d46924f92b6ca825803dce55a8471.